models/groups/index.ts

```diff
@@ -0,0 +1,77 @@
+import droleClient from '#clients/api-d-roles';
+import {
+  IDRolesGroup,
+  IDRolesGroupSearchResponse,
+} from '#clients/api-d-roles/interface';
+import { IAgentScope } from '#models/authentication/agent/scopes/constants';
+import { parseAgentScope } from '#models/authentication/agent/scopes/parse';
+import { FetchRessourceException } from '#models/exceptions';
+import { logFatalErrorInSentry } from '#utils/sentry';
+
+export class Groups {
+  /**
+   * Get all groups that a user belongs to
+   */
+  static async find(email: string): Promise<IDRolesGroupSearchResponse> {
+    try {
+      return await droleClient.getGroupsByEmail(email);
+    } catch (error) {
+      logFatalErrorInSentry(
+        new FetchRessourceException({
+          ressource: 'D-Roles Groups',
+          cause: error,
+        })
+      );
+      return [];
+    }
+  }
+
+  /**
+   * Get scopes for a specific agent from their groups
+   */
+  static async getScopeForAgent(email: string): Promise<IAgentScope[]> {
+    try {
+      const drolesGroups = await this.find(email);
+
+      const allScopesWithDuplicates = drolesGroups
+        .map((group) => {
+          const { validScopes } = parseAgentScope(group.scopes);
+          return validScopes;
+        })
+        .flat();
+
+      const uniqueScopes = [...new Set(allScopesWithDuplicates)];
+
+      return uniqueScopes;
+    } catch (error) {
+      logFatalErrorInSentry(
+        new FetchRessourceException({
+          ressource: 'D-Roles Group Scopes',
+          cause: error,
+        })
+      );
+      return [];
+    }
+  }
+
+  /**
+   * Get all groups where the user is an admin
+   */
+  static async findWhereUserIsAdmin(email: string): Promise<IDRolesGroup[]> {
+    try {
+      const groups = await this.find(email);
+      return groups.filter((group) => {
+        const user = group.users.find((user) => user.email === email);
+        return user?.is_admin ?? false;
+      });
+    } catch (error) {
+      logFatalErrorInSentry(
+        new FetchRessourceException({
+          ressource: 'D-Roles Admin Groups',
+          cause: error,
+        })
+      );
+      return [];
+    }
+  }
```

Comment:
As this is never used, should we remove it ? Less code, less tests

Reply:
Indeed, love it